stag-library/src/main/java/com/vimeo/stag/UseStag.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * The MIT License (MIT)
+ * <p/>
+ * Copyright (c) 2016 Vimeo
+ * <p/>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * <p/>
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * <p/>
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.vimeo.stag;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+/**
+ * Use this annotation to tell Stag that all the member variables of the class should be processed.
+ * <p/>
+ * If the class is annotated, Stag will generate a TypeAdapter for that class.
+ * Stag does not generate TypeAdapters for abstract classes.
+ * <p/>
+ */
+@Target({ElementType.TYPE})
+public @interface UseStag {
```

Comment:
After some discussion, this is what we would like to do for the 2.0 release. The ultimate goal would be to make using stag a one line change per model (which is what I think you are trying to drive toward):

1. Keep `@UseStag` as a class level annotation. Add options to allow all fields (excluding static and transient) in the class to be serialized (`FieldOptions.ALL`), or only those marked with `@SerializedName` (`FieldOptions.SERIALIZED_NAME`) or the deprecated `@GsonAdapterKey`

    - default `FieldOptions` will be `FieldOptions.ALL`. 
    - We will require `@UseStag` annotation in order for stag to generate a type adapter.
    - Eventually we can add more field options, but in order to reach parity with 1.x, we will just start out with the above mentioned two field options.
2. Deprecate (but keep around until a 2.1 release) `@GsonAdapterKey`
    - This will encourage moving back to `@SerializedName`. 

Do you think this works @yasirmhd @anirudhramanan @mcumings ? or do you have any name change suggestions for the field options? We can merge this PR as is and make the two changes after the merge, or if you want to make the changes in this PR that's cool too.